jsonapi/marshal.go

```diff
@@ -285,67 +287,68 @@ func getStructRelationships(relationer MarshalLinkedRelations, information Serve
 		relationships[name] = relationship
 	}
 
-	return &relationships
+	return relationships
 }
 
 // helper method to generate URL fields for `links`
 func getLinksForServerInformation(relationer MarshalLinkedRelations, name string, information ServerInformation) *Links {
-	links := &Links{}
-
-	if information != serverInformationNil {
-		prefix := strings.Trim(information.GetBaseURL(), "/")
-		namespace := strings.Trim(information.GetPrefix(), "/")
-		structType := getStructType(relationer)
+	if information == nil {
```

Comment:
I think it should be fine as `api.info` should anyways be always set to something at it is a concrete type.